### DISCUSSION TITLE
Please no unnecessary new settings

### DISCUSSION THREAD

**Discussion by Navid200** [Navid200/xDrip#20] [Ideas]
There are too many xDrip settings.

Individuals post on facebook complaining that xDrip isn't working and that they changed settings hoping to fix it, but, didn't; and now, don't remember which settings they changed.

There are experienced users who advise beginners that xDrip is too complicated.

We have a tendency of adding more and more settings.

The danger is that after we add a setting, it may be difficult to remove it, because we don't want to possibly break a system that is working.  If someone has got used to the setting being at a particular position, after removing the setting, they will have a different environment, which may confuse them.

So, it seems like we are on a downward slope (more and more settings).

May I ask the developers to scrutinize every PR and  question why a new setting is absolutely necessary?

**Comment by tzachi-dar** [comment]
I think that leaving settings out might limit our way to develop xdrip furthore.
Maybe having the settings been backed up nightly, and allowing users to return should solve the issue.

**Reply by nickrout** [reply]
It is hard to strike the right balance. xDrip+ is very complex, but that is why it is so powerful. Compare it to the Dexcom app, which is brain dead next to xDrip+.

What is really neede dis some thorough documentation that tells people in plain english what each thing does. For example, what is a Bluetooth Wixel? When do I need to enable Broadcast locally?

**Reply by Navid200** [reply]
One challenge with documentation is that there are not many individuals who know the entire capabilities of xDrip.
For example, I have created some documentation for the Dexcom side:
https://github.com/Navid200/xDrip/wiki
A lot is still missing.  But, there is no way I can add anything related to Libre as an example of something I know nothing about.

Looking at the problem from the user's perspective, I suggest that we don't allow new settings to be added just because the developer feels they need it.  I am not suggesting that we don't allow any new settings.  I suggest that we should have a requirement for approving every new setting.  The objective would be to avoid adding unnecessary settings.   The developers should be informed of this issue (why I started this thread) so that they write their code with this in mind.

I also suggest that we move the settings that are hardly ever used to the engineering mode so that by default, a new user would not have access to them.

**Reply by psonnera** [reply]
@Navid200 your wiki is becoming a reference for Dexcom, unfortunately the format is lacking a lot of flexibility (not your fault: wiki) and we can't PR a wiki should anybody want to contribute. I started something but the structure is plain wrong and the readthedocs theme sucks. If you really are motivated, we should work on a documentation approach and integrate others work like the site from @teleriddler for Libre 2. I still believe mkdocs is the easiest approach and a real documentation like AAPS with sphinx would be much better but too complex for many (myself included). Many other wikis from the xDrip+ repo should be integrated and reworked as nobody knows where they are. The documentation could be including practical aspects like your wiki and xDrip+ features explained (if we can managed to find them all as it includes so much legacy material...).

**Reply by Navid200** [reply]
@psonnera I am not sure it is such a good idea to include usage documentation in the repository.
One concern I have with xDrip is that the task has become daunting for one person.
The more we add to the repository, effectively, the more we are putting on the shoulders of the person who has to approve every pull request.

I appreciate the benefit of allowing more than one person to contribute to documentation.  But, I believe there may be other ways of accomplishing that without putting everything in one place.
For example, if there is existing documentation about Libre somewhere, we can use it as is.  

If I could use an analogy, putting everything inside the repository is like combining the department of defense and the health department and the finance department etc. of a government all together.  Those departments exist for a reason.  One of the reasons is that the expertise required in each one of those departments could be different.  Secondly, sometimes, those departments have mandates that are not necessarily in agreement.
By splitting the functions in separate departments, it becomes practical for one person to oversee and ensure that they are in coordination.
But, imagine if the head of the government was responsible for all the detail of every one of those departments.  Everything would come to a halt because that task would be humanly impossible to accomplish.

I consider the xDrip developers to be super heroes.  I want to give them a hand.  That's why I prefer not to open a PR for my documentation.  I prefer to have the documentation outside the repository.

With respect to the format, I am open to suggestions for improving it.
It is still work in progress.

**Reply by psonnera** [reply]
@Navid200 I do not intend into the xDrip+ repository, unlike wikis that are indeed linked to xDrip+ repo but not PR'able (i.e. must be submitted to xDrip+ collaborators). I also consider xDrip+ developers superheroes and wish to reduce their support efforts to the community as more the time they spend supporting less they have for developing. My idea was https://xdrip.github.io with various documenters collaborating (i.e. supervision done by xDrip+ repo contributors/collaborators but not handled by them but directly by trusted documenters). Looking again at your excellent work, it would be as hard to fit as the L2 documentation without putting at risk the main repo with DMCA like threads. At this point this would mean your model is partly right as only technical features could be documented without practical examples of the tricks implied by each option beneath xDrip+ menus and submenus. xDrip+ documentation is so fragmented that new users are driven through too many individuals efforts and a global approach would be more efficient. An idea like this. Still this can't be one individual work... Just thoughts like that.

**Reply by old-square-eyes** [reply]
> what is a Bluetooth Wixel? When do I need to enable Broadcast locally?

@nickrout a Wixel is a programmable radio unit which was used to receive G4 readings and translate them to Bluetooth for xDrip to read. Was a great solution for the G4 and had better range than a G5. It was the premiere setup and many people built little Wixel based battery powered bridge devices. I dare say it's still in use by people who use battery replaced G4 receivers. But probably EOL. I still have my bridge somewhere. Broadcast locally is to get BG from xDrip to another app like AndroaidAPS without going out over the internet. But I digress.

@Navid200 if you want help with doco, why not move it to readthedocs.io and allow Git PR's from well intentioned contributors ([much like Openaps does](url)).

**Reply by Navid200** [reply]
Allowing more than one person to add documentation expands the subjects covered, which is beneficial to the users to begin with.

However, it is inevitable that some supporters come and go.  If information is added that is outside the scope of my knowledge, I personally will not be able to maintain it.

So, those are the pros and cons from my perspective.  
The solution, in my opinion, is to have multiple people providing documentation, each for what they have expertise in.  Then, if one of those documentation sets becomes stagnant, the others will still be valid.  And the solution will be to find someone else who can address the field that requires support.

**Reply by old-square-eyes** [reply]
Things going stagnant is not a unique problem for xDrip (code or doco). I think if we are to tackle doco at all it has to be community driven with a handful of people able to sanity check. I'm no expert but it works for Openaps and AAPS. Of course a few champions do the bulk of the work, but again... that's not a unique problem. I guess it depends on what the goal is. There will never be a definitive guide/wiki. Just a leger of information, mostly up to date so people can absorb it rather than having the same questions answered inefficiently on social media. 

You can still point to other guides, as you say (e.g. Jamorham Watch Guide).

But I think community submissions is a great way to farm out the big job. There are lots of experienced people out there.

**Comment by nickrout** [comment]
I have been involved in open source for many years, I was a Linux user from around Redhat 5. 

So if I may offer some observations:

* Whatever you use it will get out of date quickly.
* User then feel unable to update as they are not quite sure how much they know.
* Facebook is a crappy support tool. It is open to most of the world, but stuff quickly gets impossible to find. There is a lot of repetition as people ask the same question over and over.
* There seem to be too many support sites for both Nightscout and xDrip+. A single definitive source would be great.

**Comment by Navid200** [comment]
I suggest we move the conversation about documentation to another thread, with a more appropriate title.  
It is an important conversation.  But, may not be easily found by someone searching for documentation because of the title.
